src/main/java/net/rptools/maptool/client/functions/DrawingFunctions.java

```diff
@@ -260,19 +368,78 @@ protected JsonObject getDrawingJSONInfo(String functionName, Zone map, GUID guid
     dinfo.addProperty("isEraser", el.getPen().isEraser() ? BigDecimal.ONE : BigDecimal.ZERO);
     dinfo.addProperty("penWidth", el.getPen().getThickness());
     dinfo.add("path", pathToJSON(d));
-    if (el.getDrawable() instanceof AbstractTemplate t) {
-      dinfo.addProperty("templateSize", t.getRadius());
+    dinfo.addProperty(
+        "isTemplate",
+        el.getDrawable() instanceof AbstractTemplate ? BigDecimal.ONE : BigDecimal.ZERO);
+    if (el.getDrawable() instanceof AbstractTemplate at) {
+      JsonObject template = new JsonObject();
+      template.addProperty("radius", at.getRadius());
+      JsonObject vertex = new JsonObject();
+      vertex.addProperty("x", at.getVertex().x);
+      vertex.addProperty("y", at.getVertex().y);
+      template.add("vertex", vertex);
+
+      // order is important here as some templates extend others, so do them first.
+      switch (at) {
+        case BlastTemplate bt -> {
+          template.addProperty("direction", bt.getDirection().toString());
+          JsonObject controlOffset = new JsonObject();
+          controlOffset.addProperty("x", bt.getOffsetX());
+          controlOffset.addProperty("y", bt.getOffsetY());
+          template.add("controlOffset", controlOffset);
+        }
+        case ConeTemplate ct -> template.addProperty("direction", ct.getDirection().toString());
+        case WallTemplate wt -> {
+          if (wt.getPath() != null) {
+            JsonArray pathInfo = new JsonArray();
+            for (CellPoint cp : wt.getPath()) {
+              ZonePoint zp = map.getGrid().convert(cp);
+              JsonObject pointInfo = new JsonObject();
+              pointInfo.addProperty("x", zp.x);
+              pointInfo.addProperty("y", zp.y);
+              pathInfo.add(pointInfo);
+            }
+            template.add("wallPath", pathInfo);
+          } else {
+            template.addProperty("wallPath", "null");
```

Comment:
A `null` path is not an expected case for `WallTemplate`. If it's possible for `WallTemplate#getPath()` to return `null` we should fix that in `WallTemplate` and not expose this case to macros.